tkg/clusterclient/clusterclient.go

```diff
@@ -2485,36 +2483,44 @@ func (c *client) UpdateAWSCNIIngressRules(clusterName, clusterNamespace string)
 	}
 
 	cniIngressRules := awsCluster.Spec.NetworkSpec.CNI.CNIIngressRules
-	// first check if existing ingress rules contains the kapp-controller port
-	for _, ingressRule := range cniIngressRules {
-		if ingressRule.Description != "kapp-controller" {
-			continue
+	cniIngressRulesNeedUpdate := false
+	for _, ingressRule := range awsIngressRules {
+		if ingressRuleIsNotPresent(ingressRule, cniIngressRules) {
+			cniIngressRules = append(cniIngressRules, ingressRule)
+			cniIngressRulesNeedUpdate = true
 		}
+	}
 
-		if ingressRule.Protocol != capav1beta2.SecurityGroupProtocolTCP {
-			continue
+	if cniIngressRulesNeedUpdate {
+
+		awsCluster.Spec.NetworkSpec.CNI.CNIIngressRules = cniIngressRules
+		if err := c.UpdateResource(awsCluster, clusterName, clusterNamespace); err != nil {
+			return err
 		}
+	}
 
-		if ingressRule.FromPort != DefaultKappControllerHostPort || ingressRule.ToPort != DefaultKappControllerHostPort {
+	return nil
+}
+
+func ingressRuleIsNotPresent(rule capav1beta2.CNIIngressRule, listOfRules capav1beta2.CNIIngressRules) bool {
+	for _, ingressRule := range listOfRules {
+		if ingressRule.Description != rule.Description {
 			continue
 		}
 
-		return nil
-	}
-
-	cniIngressRules = append(cniIngressRules, capav1beta2.CNIIngressRule{
-		Description: "kapp-controller",
-		Protocol:    capav1beta2.SecurityGroupProtocolTCP,
-		FromPort:    DefaultKappControllerHostPort,
-		ToPort:      DefaultKappControllerHostPort,
-	})
+		if ingressRule.Protocol != rule.Protocol {
+			continue
+		}
 
-	awsCluster.Spec.NetworkSpec.CNI.CNIIngressRules = cniIngressRules
-	if err := c.UpdateResource(awsCluster, clusterName, clusterNamespace); err != nil {
-		return err
+		if ingressRule.FromPort != rule.FromPort {
+			continue
+		}
+		if ingressRule.ToPort != rule.ToPort {
+			continue
+		}
+		return false
 	}
-
-	return nil
+	return true
 }
```

Comment:
This  UpdateAWSCNIIngressRules  is called by handleKappControllerUpgrade, probably should change that.

And also the description of func UpdateAWSCNIIngressRules is for kapp now

Reply:
?? this is ingressRuleIsNotPresent. Sorry I don't follow the comment.

Reply:
modified comment

Reply:
Sorry for the confusion. What I mean is, this func UpdateAWSCNIIngressRules was executed only for kapp controller special handling(in func handleKappControllerUpgrade).

Now it contains ports handling more than just for kapp controller, so we probably should change the func name of  handleKappControllerUpgrade also

https://github.com/vmware-tanzu/tanzu-framework/blob/647899d9d455f3e131487e9a5dac36c38d2c969f/tkg/client/upgrade_cluster.go#L1174-L1195

Reply:
Is changing the function name necessary ?

Reply:
Logically should be fine but this func was designed only for kapp-controller, so the corresponding descriptions are all points to the kapp-controller

https://github.com/vmware-tanzu/tanzu-framework/blob/647899d9d455f3e131487e9a5dac36c38d2c969f/tkg/client/upgrade_cluster.go#L416-L419

Reply:
I think is named the correct thing: 
UpdateAWSCNIIngressRules
which is what it does :) why you call it. That's up to the caller right. 
Its a generic function you can call on an awscluster
What we should take out is the comments. ... which are always best left off when possible. The name of the function UpdateAWSCNIIngressRules is pretty accurate, and the comments which show up on an IDE also state what it does.  More generic functions are always better I think.

Reply:
I see your point...Ok, I am resolving this, thanks

Reply:
IMHO,  The function is public, which means it can be access by anything, that needs to make sure the correct ingress rules are set. 
We could make it so it has to be called for each ingress rule, let me finish this pr and start a new one with those changes just to have something that solves the problem asap.

Reply:
I started https://github.com/vmware-tanzu/tanzu-framework/pull/4086 which is an approach that handles kapp-controller and addons-manager separately.